src/test/java/com/google/cloud/tools/managedcloudsdk/ManagedCloudSdkIntegrationTest.java

```diff
@@ -0,0 +1,145 @@
+    componentFuture.get();
+
+    Assert.assertTrue(testSdk.isInstalled());
+    Assert.assertTrue(testSdk.hasComponent(testComponent));
+    Assert.assertTrue(testSdk.isUpToDate());
+
+    // make sure we cant update a versioned cloud sdk
+    Assert.assertEquals(null, testSdk.newUpdater());
+  }
+
+  @Test
+  public void testManagedCloudSdk_latestInstall()
+      throws BadCloudSdkVersionException, UnsupportedOsException, ExecutionException, IOException,
+          CommandExitException, InterruptedException {
+    ManagedCloudSdk testSdk =
+        new ManagedCloudSdk(Version.LATEST, tempDir.getRoot().toPath(), OsInfo.getSystemOsInfo());
+
+    Assert.assertFalse(testSdk.isInstalled());
+    Assert.assertFalse(testSdk.hasComponent(testComponent));
+    Assert.assertFalse(testSdk.isUpToDate());
+
+    ListenableFuture<Path> installFuture = testSdk.newInstaller().downloadSdk(testListener);
+    installFuture.get();
+
+    Assert.assertTrue(testSdk.isInstalled());
+    Assert.assertFalse(testSdk.hasComponent(testComponent));
+    Assert.assertTrue(testSdk.isUpToDate());
+
+    ListenableFuture<Void> componentFuture =
+        testSdk.newComponentInstaller().installComponent(testComponent, testListener);
+    componentFuture.get();
+
+    Assert.assertTrue(testSdk.isInstalled());
+    Assert.assertTrue(testSdk.hasComponent(testComponent));
+    Assert.assertTrue(testSdk.isUpToDate());
+  }
+
+  @Test
+  public void testManagedCloudSdk_updateLatest()
+      throws ExecutionException, IOException, CommandExitException, UnsupportedOsException,
+          InterruptedException {
+    ManagedCloudSdk testSdk =
+        new ManagedCloudSdk(Version.LATEST, tempDir.getRoot().toPath(), OsInfo.getSystemOsInfo());
+
+    Assert.assertFalse(testSdk.isInstalled());
+    Assert.assertFalse(testSdk.isUpToDate());
+
+    ListenableFuture<Path> installFuture = testSdk.newInstaller().downloadSdk(testListener);
+    installFuture.get();
+
+    Assert.assertTrue(testSdk.isInstalled());
+    Assert.assertTrue(testSdk.isUpToDate());
+
+    // forcibly downgrade the cloud SDK so we can test updating.
+    new CommandFactory()
+        .newRunner(
+            Arrays.asList(
+                testSdk.getGcloud().toString(),
+                "components",
+                "update",
+                "--quiet",
+                "--version=" + FIXED_VERSION),
+            null,
+            null,
+            testListener)
+        .run();
+
+    Assert.assertTrue(testSdk.isInstalled());
+    Assert.assertFalse(testSdk.isUpToDate());
+
+    ListenableFuture<Void> updateFuture = testSdk.newUpdater().update(testListener);
+    updateFuture.get();
+
+    Assert.assertTrue(testSdk.isInstalled());
+    Assert.assertTrue(testSdk.isUpToDate());
+  }
+
+  static class StdOutMessageListener implements MessageListener {
+    @Override
+    public void message(String rawString) {
+      System.out.print(rawString);
```

Comment:
General principle that passing tests should generate no output. Consider using a non stdout PrintStream instead.